client/src/components/StreamField/blocks/FieldBlock.js

```diff
@@ -122,6 +122,12 @@ export class FieldBlock {
       .querySelectorAll('.error-message')
       .forEach((element) => element.remove());
 
+    // The widget knows exactly where the <input> is, so we let it handle the
+    // invalid state to e.g. set the aria-invalid attribute. Use optional
+    // chaining as custom widgets may not have implemented this method, or the
+    // widget itself may have failed to render in the first place.
+    this.widget?.setInvalid?.(!!error);
```

Comment:
I considered passing the `error` as-is and maybe calling the method `setError` instead coercing it to boolean and calling the method `setInvalid`, in case the widget wants to do something smart with the error messages.

However, it seems the error message is already handled by the `FieldBlock` rather than the widget (as seen in a few lines below this), so I opted to just keep it simple. Furthermore, there's already `setError` method in the `FieldBlock` itself, so I think the distinction helps.